### ISSUE TITLE
Support for whitelisted domains in a Dockerfile?

### ISSUE THREAD

**Issue by nic-b** [hadolint/hadolint#309]
- [ ] This is a bug report
- [x] This is a feature request
- [x] I searched existing issues before opening this one

Hi,

Just started looking at hadolint, specifically around its functionality to enforce/lint trusted repositories for images.

Taking this rule one step further, is it possible to apply a similar concept to domains referenced within an install command, within `RUN`? If not, would it be something  that could be contributed back?

```Dockerfile - don't allow this
RUN pip install --trusted-host pypi.python.org -r requirements.txt
```

```Dockerfile - allow this
RUN pip install --trusted-host mycompany.com -r requirements.txt
```

Using the example above, my use case would be that within an organisation, we'd like to restrict any sources for installs, to (trusted) internal domains only. In the same way the hadolint config file allows input for `trustedRegistries`, there would be config for `trustedDomains`.

Any thoughts or help appreciated
thanks

**Comment by lorenzo** [comment]
There is currently no such feature, but looks like a great idea. 

would this feature only affect pip? Or what other use cases do you have in mind?

**Comment by nic-b** [comment]
I'm mostly concerned about commands that can install code from configurable sources - pip install, npm install, go get, to name but a few. Even curl/wget to an extent. However I could easily see a use case for checking all URL's within the entire dockerfile and only allowing whitelisted/disallow blacklisted.

**Comment by lorenzo** [comment]
I guess we could check for urls used in known programs, but a generic solutions seems difficult.

Would you like to put some ideas together with a proposal on how this would work?

**Comment by nic-b** [comment]
thanks for getting back to me. Sure. I'll take a look a the source (dont know Haskell that well mind) and come up with some ideas.

**Comment by lorenzo** [comment]
More that ideas for how to code this, I was looking for examples where rules would pass/fail

**Comment by nic-b** [comment]
Ok. I was thinking just more of the same as illustrated by the pip install command above.

With the examples below, you could have config like:

```
trustedDomains:
  - mygoodurl.com
  - mygoodurl.net
```

```
trustedDomains:
  - mygoodurl.*
```

```
trustedDomains:
  - my*url.com
```



**_Dockerfile examples_**

**Simple trusted host url for pip install**
`RUN pip install --trusted-host mybadurl.com -r requirements.txt`

**Defining the pip.conf on the fly to redirect to a specified url**
```
RUN mkdir -p config/pip.conf; \
           echo "[global]" >> config/pip.conf; \
           echo "index-url = https://mybadurl.com" >>config/pip.conf; \
           echo "extra-index-url = " >>config/pip.conf; \
           echo "       https://mygoodurl.net" >>config/pip.conf; \
RUN pip install -y boto3
```

**Any url specified in curl**
`RUN curl -SL https://mybadurl.com --output badurl.tar.gz`

**Any url specified in wget**
`RUN wget https://mybadurl.com/daneel.tar`

**Any variable usage with urls in, shown with curl as an example**
```
ENV MY_URL=https://mybadurl.com
RUN curl -SL $MY_URL --output badurl.tar.gz
```

**New repo add for yum and subsequent install**
```
yum-config-manager --add-repo="https://badurl.com"
yum install httpd  -y
```
**npm install**
`npn install https://mybadurl.com`

**Comment by nic-b** [comment]
Where there any thoughts on the above and whether this is something that could be looked at, or even contributed back?
